datajoint/base_relation.py

```diff
@@ -289,27 +289,35 @@ def check_fields(fields):
                 else:
                     raise
 
-    def delete_quick(self):
+    def delete_quick(self, get_count=False):
         """
-        Deletes the table without cascading and without user prompt. If this table has any dependent
-        table(s), this will fail.
+        Deletes the table without cascading and without user prompt.
+        If this table has populated dependent tables, this will fail.
         """
         query = 'DELETE FROM ' + self.full_table_name + self.where_clause
         self.connection.query(query)
+        count = self.connection.query("SELECT ROW_COUNT()").fetchone()[0] if get_count else None
         self._log(query[:255])
+        return count
 
-    def delete(self):
+    def delete(self, verbose=True):
         """
         Deletes the contents of the table and its dependent tables, recursively.
         User is prompted for confirmation if config['safemode'] is set to True.
         """
+        already_in_transaction = self.connection.in_transaction
+        safe = config['safemode']
+        if already_in_transaction and safe:
```

Comment:
What's the rationale for disallowing deletion when already in transaction if in safe mode? This differs from previous behavior as far as I recall.

Reply:
Because there is no way to under the delete in the middle of a transaction without rolling back the entire transaction.  So there is no *safe* delete within a transaction.

Reply:
Sounds fair

Reply:
Sounds fair